spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/VaultEnvironmentRepository.java

```diff
@@ -116,6 +124,13 @@ public VaultEnvironmentRepository(ObjectProvider<HttpServletRequest> request,
 
 		this.accessStrategy = VaultKvAccessStrategyFactory.forVersion(rest, baseUrl,
 				properties.getKvVersion());
+
+		if (tokenProvider == null) {
```

Comment:
Instead of passing `null` in the old constructor why not just pass a new `HttpRequestConfigTokenProvider`?

Reply:
You're right, that's better. I'll update it.

Reply:
Done